common/src/main/java/com/wynntils/screens/maps/MainMapScreen.java

```diff
@@ -164,8 +168,14 @@ protected void doInit() {
                         return;
                     }
 
-                    if (Models.Compass.getCompassLocation().isPresent()) {
-                        Location location = Models.Compass.getCompassLocation().get();
+                    List<MarkerInfo> markers = Models.Marker.USER_WAYPOINTS_PROVIDER
+                            .getCompassInfos()
+                            .toList();
+                    if (!markers.isEmpty()) {
+                        int index = markers.indexOf(focusedMarker);
```

Comment:
what if indexOf fails? Maybe have a safety check for -1, even if that "is not supposed to happen".

Reply:
-1 + 1 = 0 = reset the counter. Perfect behavior. I can add a comment for it..

Reply:
But then we have to check if it is in bounds. I think this is a bit cleaner. But this whole screen needs adaptation, this is just to get it working somewhat reasonably.

Reply:
Oh. Very magic. Not sure I like it. But sure, as long as you write a comment.

Reply:
Done